### PR TITLE
add ld-cmp-jcc global stats

### PR DESCRIPTION
**Pull request by amiri-khalil**

### CODE REVIEW

File: lbr.py [hunk 1]
```diff
@@ -508,8 +509,14 @@ def f2s(x): return C.flag2str(' ', C.env2str(x, prefix=True))
           if edge_en and is_type(x86.COND_BR, line):
             if is_taken(line): glob['cond_taken-not-first'] += 1
             else: glob['cond_non-taken'] += 1
-            if x86.is_fusion(lines[-1], line): glob['cond_fusible'] += 1
-            else: glob['cond_non-fusible'] += 1
+            if x86.is_fusion(lines[-1], line):
+              glob['cond_fusible'] += 1
+              if is_type(x86.TEST_CMP, lines[-1]) and is_type(x86.LOAD, lines[-2]):
```
Comment (aayasin) [inline lbr.py:514]: This needs to check lines already has 2 instructions to be safe

File: lbr.py [hunk 2]
```diff
@@ -660,7 +667,7 @@ def nm(x):
     return n
   if len(comment): comment = '\t:(see %s below)' % c(comment)
   elif ratio_of: comment = '\t: %7s of %s' % (ratio(count, ratio_of[1]), ratio_of[0])
-  print('%s of %s: %10s%s' % (c(prefix), '{: >{}}'.format(c(nm(name)), 45 - len(prefix)), str(count), comment))
+  print('%s of %s: %10s%s' % (c(prefix), '{: >{}}'.format(c(nm(name)), 60 - len(prefix)), str(count), comment))
```
Comment (aayasin) [inline lbr.py:670]: Better calculate the longest stat from Insts_Global[] to avoid manual update next time

File: lbr.py [hunk 3]
```diff
@@ -691,11 +699,14 @@ def nc(x): return 'non-cold ' + x
                    prefix='misprediction-ratio', comment='paths histogram')
 
 def print_common(total):
+  def print_notes():
+    print('# CMP denotes cmp or test instructions\n')
```
Comment (aayasin) [inline lbr.py:703]: Upper case CMP or TEST

File: lbr.py [hunk 4]
```diff
@@ -691,11 +699,14 @@ def nc(x): return 'non-cold ' + x
...
   if glob['size_stats_en']:
     totalv = (total - stat['bad'] - stat['bogus'])
     stat['size']['avg'] = round(stat['size']['sum'] / totalv, 1) if totalv else -1
   print('LBR samples:', hist_fmt(stat))
   if edge_en and total: print_global_stats()
+  print_notes()
   print('#Global-stats-end\n')
```
Comment (aayasin) [inline lbr.py:710]: How about denser one call of:
print('\n'.join(["# CMP ..", "#Global-stats-end", ""])

File: kernels/x86.py [hunk 5]
```diff
@@ -24,6 +24,8 @@
 INDIRECT  = r"(jmp|call).*%"
 CALL_RET  = '(call|ret)'
 COND_BR   = 'j[^m][^ ]*'
+TEST_CMP  = r"(test|cmp).?\s"
+LOAD      = r"mov.?\s.*\).*,"
```
Comment (aayasin) [inline kernels/x86.py:28]: I suspect this may catch stores as well, but it does not seem to be used by this commit

**Review by aayasin — CHANGES_REQUESTED** [review]
Adjust per enclosed review

File: kernels/x86.py (hunk 5 above)
Comment (amiri-khalil) [reply kernels/x86.py:28]: I think the ',' after ')' makes it a load definitely

File: lbr.py (hunk 4 above)
Comment (amiri-khalil) [reply lbr.py:710]: I used this so we can add notes easily in future commits to print_notes()

**Pull request merged**